src/components/Editor/CodeEditor.tsx

```diff
@@ -0,0 +1,211 @@
+      editorDidMount={(editor) => {
+        const updateCursor = (clientId: string, pos: number) => {
+          const clientCursor = otherClientsCursor.current.get(clientId);
+          clientCursor!.updateCursor(editor, pos);
+        };
+
+        const updateLine = (clientId: string, fromIdx: number, toIdx: number) => {
+          const clientCursor = otherClientsCursor.current.get(clientId);
+          clientCursor!.updateLine(editor, fromIdx, toIdx);
+        };
+
+        doc?.subscribe((event: any) => {
+          if (event.name === 'remote-change') {
+            event.value.forEach((change: any) => {
+              const actor = change.id.actor;
+              if (actor !== client.getID()) {
+                if (!otherClientsCursor.current.has(actor)) {
+                  connectClient(actor);
+                  updateCursor(actor, 0);
+                  // TODO Load user's cursor position
+                }
+              }
+            });
+          }
+        });
+
+        const root = doc.getRootObject() as any;
+        root.content.onChanges((changes: any) => {
+          for (const change of changes) {
+            const actor = change.actor;
+            const from = change.from;
+            const to = change.to;
+            if (change.type === 'content') {
+              const content = change.content || '';
+
+              if (actor !== client.getID()) {
+                console.log(`%c remote: ${from}-${to}: ${content}`, 'color: skyblue');
+                const fromIdx = editor.posFromIndex(from);
+                const toIdx = editor.posFromIndex(to);
+                editor.replaceRange(content, fromIdx, toIdx, 'yorkie');
+              }
+            } else if (change.type === 'selection') {
+              if (actor !== client.getID()) {
+                if (!otherClientsCursor.current.has(actor)) {
+                  connectClient(actor);
+                }
+
+                const fromIdx = editor.posFromIndex(from);
+                const toIdx = editor.posFromIndex(to);
+                updateCursor(actor, change.to);
+                updateLine(actor, fromIdx, toIdx);
+              }
+            }
+          }
+        });
+
+        // We need to subtract the height of NavBar.
+        editor.setSize('auto', 'calc(100vh - 64px)');
+        editor.setValue(root.content.getValue());
+      }}
+      // Notifying other clients to move the cursor
+      onSelection={(editor, data) => {
+        if (data.origin === undefined) {
+          return;
+        }
+
+        let from: number;
+        let to: number;
+        from = editor.indexFromPos(data.ranges[0].anchor);
+        to = editor.indexFromPos(data.ranges[0].head);
+
+        if (from > to) {
+          [from, to] = [to, from];
+        }
+        doc?.update((root: any) => {
+          root.content.updateSelection(from, to);
+        });
+      }}
+      onBeforeChange={(editor: any, change: any) => {
+        console.log(change.origin, change.text);
```

Comment:
Is there any reason to output this data?

Reply:
https://github.com/yorkie-team/yorkie-codepair/blob/master/src/components/CodeEditor.tsx#L99
The previously printed part was imported. I thought about deleting it, but I left it as it is. 

I think it would be better to erase it!

Reply:
I think it would be nice to erase it. By the way, it would be nice to add linter to this project in the next PR.